syllabus/core/environment_sync_wrapper.py

```diff
@@ -74,23 +77,28 @@ def step(self, action):
             self.step_updates.append({
                 "update_type": "step",
                 "metrics": (obs, rew, term, trunc, info),
+                "env_id": self.instance_id,
                 "request_sample": False
             })
             # Task progress
             self.step_updates.append({
                 "update_type": "task_progress",
                 "metrics": ((self.task_space.encode(self.env.task), self.task_progress)),
+                "env_id": self.instance_id,
                 "request_sample": term or trunc
             })
             # Send batched updates
-            if len(self.step_updates) >= 1000 or term or trunc:
-                self.update_queue.put(self.step_updates)
+            if len(self.step_updates) >= 100 or term or trunc:
+                # Group updates into arrays
+                updates = self._aggregate_step_updates()
```

Comment:
No reason to package dicts then repackage into nparrays. We can buffer 100 steps into an nparray directly

Reply:
Fixed